### PR TITLE
Feature/acl impl datasource and conversation

### PR DESCRIPTION
**Pull request by uma3355**

### CODE REVIEW

**Review by CodeCrafter7 — COMMENTED** [review]
@uma3355 Cant merge this right now because this will break the FE, the chat list will not render

File: api/src/app/database/models.py [hunk 1]
```diff
@@ -226,6 +226,7 @@ class DataSourceEntity(Base):
     schemas = Column(ARRAY(String))
     tables_to_include = Column(ARRAY(String))
     custom_schema_description = Column(Text)
+    acl = Column(JSONB)
```
Comment (CodeCrafter7) [inline api/src/app/database/models.py:229]: @uma3355 I think this is not required anymore

File: api/src/app/database/repository.py [hunk 2]
```diff
@@ -406,6 +407,57 @@ def get_data_source_by_id(id):
             logging.error(f"Exception while getting data source: {ex}")
         finally:
             session.close()
+        
+    def get_all_data_source(userName, userGroups, userRoles):
+        try:
+            session=Session()
+            subquery = Persistence.acl_subquery(userName, userGroups, userRoles, 'data_source', session)
```
Comment (CodeCrafter7) [inline api/src/app/database/repository.py:414]: @uma3355 can you please remove the print statement, also try to do a single query using a join?

**Review by CodeCrafter7 — CHANGES_REQUESTED** [review]

File: api/src/app/repo/db.py [hunk 3]
```diff
@@ -38,18 +42,64 @@ def get_filter_parameter(filter_list):
...
+            "owned": [
+                match_query,project_stage
+            ]
+        }
+    }
+    if flag== None:
+        facet_stage["$facet"]["shared"][0]["$match"].pop("archived", None)
+
+
+    print(facet_stage)
```
Comment (CodeCrafter7) [inline api/src/app/repo/db.py:82]: @uma3355 remove print statement

File: api/src/app/repo/db.py [hunk 4]
```diff
@@ -38,18 +42,64 @@ def get_filter_parameter(filter_list):
...
 
+def conversation_pipeline(user_id: Union[ObjectId, str], flag, username, user_groups, user_roles):
+    match_query = {}
+    
+    if flag== None:
+        match_query["$match"] = {"_id": user_id, "acl.owner": username}
+    elif isinstance(user_id, str) and flag != None:
+        match_query["$match"] = {"user_id": user_id, "archived": flag, "acl.owner": username}
+    
+    print(match_query)
```
Comment (CodeCrafter7) [inline api/src/app/repo/db.py:53]: remove print

**Review by CodeCrafter7 — CHANGES_REQUESTED** [review]

**Comment by uma3355** [comment]
this code is in feature/conversation-acl-update3 branch

**Pull request closed without merging**